modo/api.py

```diff
@@ -370,3 +377,28 @@ def enrich_metadata(self):
                     self.update_element(inst_names[ele.name], ele)
                 else:
                     continue
+
+    def stream_cram(
+        self, cram_path: str, region: str = None, output: str = None
+    ):
+        """Slices and streams the requested CRAM file, both local and remote,
+        and either outputs a data stream or writes data to local file"""
+
+        # check requested CRAM exists in MODO
+        if Path(cram_path) not in self.list_files():
+            raise ValueError(f"{cram_path} not found in {self.path}.")
+
+        if self.s3_endpoint:
+            # http://domain/s3 + bucket/modo/file.cram --> http://domain/htsget/reads/modo/file.cram
+            url = (
+                self.htsget_endpoint
+                + "/reads/"
+                + str(Path(*Path(cram_path).parts[1:]))
+            )
+            # str(Path(*Path(cram_path).parts[1:])) same as path.split("/", maxsplit=1)[1] but cross-platform
+            slice_remote_cram(url, region, output)
```

Comment:
__Question__: For local slicing we return a `pysam.libcalignmentfile.IteratorRowRegion`, but for remote slicing we return the stream. I think it would be good to have a consistent behavior for remote and local streaming. Is it possible to transform the stream into a similar pysam rowregion as well?

Reply:
To harmonize method behavior I split stream_cram() into two methods:
- stream_cram() - streams or returns an iterator (in the future we can further harmonize this)
- save_cram() - saves the output to a local file.
(see commit - https://github.com/sdsc-ordes/modo-api/commit/0b0ed93e2029420d128e1c81b0ce374ec7b128f3)

Reply:
stream_cram() harmonized as well, returns iterator in both local and remote cases.